pkg/apis/seedmanagement/validation/managedseed.go

```diff
@@ -45,6 +48,46 @@ func ValidateManagedSeed(managedSeed *seedmanagement.ManagedSeed) field.ErrorLis
 
 	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&managedSeed.ObjectMeta, true, apivalidation.NameIsDNSLabel, field.NewPath("metadata"))...)
 	allErrs = append(allErrs, ValidateManagedSeedSpec(&managedSeed.Spec, field.NewPath("spec"), false)...)
+	allErrs = append(allErrs, ValidateHAConfig(managedSeed)...)
+
+	return allErrs
+}
+
+// ValidateHAConfig validates the HA configuration for the ManagedSeed object.
+func ValidateHAConfig(managedSeed *seedmanagement.ManagedSeed) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// extract the SeedTemplate
+	seedTemplate, seedTemplatePath, err := helper.ExtractSeedTemplate(managedSeed)
+	if err != nil {
+		return append(allErrs, field.Invalid(seedTemplatePath, managedSeed.Spec, fmt.Sprintf("failed to extract SeedTemplate %v", err)))
+	}
+
+	multiZonalSeedLabelPath := seedTemplatePath.Child("metadata", "labels").Key(v1beta1constants.LabelSeedMultiZonal)
+	seedSpecHAPath := seedTemplatePath.Child("spec", "highAvailability")
+	allErrs = append(allErrs, validateManagedSeedShouldNotHaveBothMultiZoneLabelAndHASpec(seedTemplate.ObjectMeta, &seedTemplate.Spec, multiZonalSeedLabelPath, seedSpecHAPath)...)
+
+	// validate the value of label if present.
+	if multiZoneLabelVal, ok := seedTemplate.ObjectMeta.Labels[v1beta1constants.LabelSeedMultiZonal]; ok {
```

Comment:
```suggestion
	if multiZoneLabelVal := seedTemplate.ObjectMeta.Labels[v1beta1constants.LabelSeedMultiZonal]; multiZoneLabelVal != "" {
```

Reply:
Shoudn't we check if the label is present and then validate if the value present is a valid boolean value or empty?

Reply:
done